service/src/main/kotlin/xyz/lebkuchenfm/domain/songs/SongsService.kt

```diff
@@ -1,7 +1,36 @@
 package xyz.lebkuchenfm.domain.songs
 
-class SongsService(private val repository: SongsRepository) {
+import com.github.michaelbull.result.get
+import xyz.lebkuchenfm.external.youtube.YoutubeClient
+
+class SongsService(private val repository: SongsRepository, private val youtubeClient: YoutubeClient) {
     suspend fun getAllSongs(): List<Song> {
         return repository.findAllOrderByNameAsc()
     }
+
+    suspend fun incrementPlayCount(youtubeId: String): Song? {
```

Comment:
Why `youtubeId`? Just use the `Song` instance, or at least `name`.

Reply:
When w use `q` and its argument as a name is not found in database then we are getting song from youtube and there id is const but name/title can be changed overtime. I thought it would be safer to use youtube id for that particular reason.